packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -320,6 +322,55 @@ describe('<DataGrid /> - Pagination', () => {
         );
         expect(getColumnValues(0)).to.deep.equal(['7', '8']);
       });
+
+      it('should update the amount of rows rendered and call onPageSizeChange when changing the table height', async () => {
+        const onPageSizeChange = spy();
+
+        const TestCaseAutoPageSize = (props: { nbRows: number; height?: number }) => {
+          const data = useData(props.nbRows, 10);
+
+          return (
+            <div style={{ width: 300, height: props.height }}>
+              <DataGrid
+                columns={data.columns}
+                rows={data.rows}
+                autoPageSize
+                pagination
+                onPageSizeChange={onPageSizeChange}
+              />
+            </div>
+          );
+        };
+        const nbRows = 27;
+        const heightBefore = 780;
+        const heightAfter = 360;
+
+        const { setProps } = render(<TestCaseAutoPageSize nbRows={nbRows} height={heightBefore} />);
+
+        const footerHeight = document.querySelector('.MuiDataGrid-footerContainer')!.clientHeight;
+        const expectedViewportRowsLengthBefore = Math.floor(
+          (heightBefore - DEFAULT_GRID_OPTIONS.headerHeight - footerHeight) /
+            DEFAULT_GRID_OPTIONS.rowHeight,
+        );
+        const expectedViewportRowsLengthAfter = Math.floor(
+          (heightAfter - DEFAULT_GRID_OPTIONS.headerHeight - footerHeight) /
+            DEFAULT_GRID_OPTIONS.rowHeight,
+        );
+
+        let rows = document.querySelectorAll('.MuiDataGrid-viewport [role="row"]');
+        expect(rows.length).to.equal(expectedViewportRowsLengthBefore);
+
+        setProps({ height: heightAfter });
+
+        await waitFor(() => getCell(expectedViewportRowsLengthAfter, 0));
```

Comment:
You could wait for the "1-5 of 100" text to change. It would be closer to what the user sees.